include/soci/soci-backend.h

```diff
@@ -22,8 +22,19 @@ namespace soci
 // data types, as seen by the user
 enum data_type
 {
-    dt_string, dt_date, dt_double, dt_integer, dt_long_long, dt_unsigned_long_long,
```

Comment:
Sorry, this is a major problem IMO: we need to keep `dt_integer` and `dt_[unsigned_]long_long` names as removing them would break existing code using SOCI (see e.g. [this project](https://cs.github.com/rstudio/rstudio/blob/2a3eead88a128805c249a6aa7a3dbff8e9ae9faa/src/cpp/session/modules/zotero/ZoteroCollectionsLocal.cpp#L88) for a random example).

For `dt_long_long` it's pretty simple as it can be just a synonym for `dt_int64`. For `dt_integer` it might be more complicated, but it needs to be mapped to `dt_int32` or `dt_int64` depending on the actual `sizeof(int)`. But it still needs to exist and continue to work.

Reply:
Good point. Internally though, I would prefer to exclusively use the fixed-size names (dt_int32, dt_int64 etc.). Do you have a different opinion on that?

Reply:
I'd prefer not to change the tests, to avoid the code churn and to make sure the old constants still work, but using fixed width types in the implementation Indeed seems like a good idea.

Reply:
> For dt_integer it might be more complicated, but it needs to be mapped to dt_int32 or dt_int64 depending on the actual sizeof(int).

If we had a `dt_long` before, I would understand your concern. But a `dt_integer` a.k.a. `int` is never 64-bit and therefore should always be mapped to `dt_int32`, shouldn't it?

Reply:
I think there are ILP64 platforms, where int is 64 bit, but maybe it's indeed an overkill to worry about those as I've never heard of SOCI being used on them, so I agree, let's just make it a synonym for `dt_int32` to keep things simple, sorry for overthinking this initially.